CODEWALK.md

```diff
@@ -314,6 +313,111 @@ Protocols (goroutines) communicate through custom channels.
 This architecture needs more thought, but we are considering moving more inter-module communication to use iterators (c.f. those in Java). 
 An event bus might also be a good pattern for some cases, though.
 
+## libp2p networking
+
+Filecoin relies on [libp2p](https://libp2p.io/) for its networking needs.
+
+libp2p is a modular networking stack for the peer-to-peer era. It offers building blocks to tackle requirements such as
+peer discovery, transport switching, multiplexing, content routing, NAT traversal, pubsub, circuit relay, etc., most of which Filecoin uses.
+Developers can compose these blocks easily to build the networking layer behind their P2P system.
+
+Here we'll explain some of the mechanics that drive the Filecoin network layer, in particular:
+
+1. Peer discovery.
+2. NAT topology determination.
+3. Autorelay services for solving connectivity issues.
+
+In addition, it must be noted that Filecoin relies on [GossipSub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub) for 
+pubsub gossip among peers propagating blockchain blocks and unmined messages.
+
+### Peer discovery
+
+We use the libp2p Kademlia Distributed Hashtable (Kad DHT) service to:
+
+1. discover Filecoin peers to interface with upon boot.
```

Comment:
filecoin usually calls them "nodes" (also imperfect bc this doc talks so much about the node API). maybe give a heads up on the synonym with something like "Filecoin nodes, aka peers"